Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -681,13 +689,16 @@ extension HTTPClient {
 
         @discardableResult
         func setConnection(_ connection: Connection) -> Connection {
```

Comment:
not related to this PR, but why do we also return the connection that we set here?